pydynamo/storage/disk/sstableiterator.py

```diff
@@ -0,0 +1,230 @@
+from ..error import StorageException
+from ..error import ErrorType
+from ..iterator import Iterator
+from struct import unpack, pack
+import os
+
+
+class SStableIterator(Iterator):
+    INT_SIZE = 4
+
+    def __init__(self, store_name: str,
+                 id: int, size: int, index_table, path, last_index) -> None:
+        """
+        :param store_name: the name of store the sstable belongs to
+        :param id: the id of the sstable
+        :param size: the size of sstable file
+        :param index_table: the index table stored in memory for this sstable
+        :param path: the path the sstable file and index file store in
+        :param last_index: the offset of last
+        key-data-time set data in sstable file
+        """
+        sstable_file_name = store_name + str(id) + ".db"
+        index_file_name = store_name + str(id) + ".index"
+        self.cur = None
+        sstable_dir = os.path.join(path, "sstable")
+        index_dir = os.path.join(path, "index")
+        self.sstable_file = open(os.path.join(sstable_dir, sstable_file_name),
+                                 "rb")
+        self.index_file = open(os.path.join(index_dir, index_file_name), "rb")
+        self.index_table = index_table
+        self.cur_offset = -1
+        self.file_size = size
+        self.last_index = last_index
+
+    def __del__(self) -> None:
+        """
+        close sstable file and index file
+        :return: None
+        """
+        self.sstable_file.close()
+        self.index_file.close()
+
+    def seek(self, key: str) -> None:
+        """
+        :param key: the key the iterator will point to
+        :return:
+        """
+        index_range = self._get_range(key)
+        if len(index_range) == 1:
+            offset = unpack("i", index_range[0])[0]
+            self.cur_offset = offset
+        elif len(index_range) == 2:
+            offset = self._get_offset(key, index_range[0], index_range[1])
+            if offset is None:
+                raise StorageException(ErrorType.NOT_FOUND,
+                                       "The key is not found in table.")
+            self.cur_offset = offset
+
+    def seek_to_first(self) -> None:
+        """
+        set the iterator point to the beginning
+        :return: None
+        """
+        self.cur_offset = -1
+
+    def valid(self) -> bool:
+        """
+        :return: true if the iterator has next. Otherwise false
+        """
+        if self.cur_offset == -1:
+            return True
+        elif self.cur_offset < -1:
+            return False
+        else:
+            self.sstable_file.seek(self.cur_offset)
```

Comment:
Few comments:
1. if you hit the end of sstable; seems this function will throw exception instead of returning False
2. please add unit tests for Iterator itself.